packages/navigator/src/browser/open-editors-widget/navigator-open-editors-widget.tsx

```diff
@@ -0,0 +1,178 @@
+ ********************************************************************************/
+
+import * as React from '@theia/core/shared/react';
+import { injectable, interfaces, Container, postConstruct, inject } from '@theia/core/shared/inversify';
+import {
+    ApplicationShell,
+    ContextMenuRenderer,
+    defaultTreeProps,
+    NodeProps,
+    TreeDecoratorService,
+    TreeModel,
+    TreeNode,
+    TreeProps,
+    TREE_NODE_CONTENT_CLASS,
+} from '@theia/core/lib/browser';
+import { OpenEditorNode, OpenEditorsModel } from './navigator-open-editors-tree-model';
+import { createFileTreeContainer, FileTreeModel, FileTreeWidget } from '@theia/filesystem/lib/browser';
+import { OpenEditorsTreeDecoratorService } from './navigator-open-editors-decorator-service';
+import { OpenEditorTreeDecorationData } from './navigator-open-editors-file-decorator';
+import { notEmpty } from '@theia/core/lib/common';
+import { OPEN_EDITORS_CONTEXT_MENU } from './navigator-open-editors-menus';
+import { EditorPreviewWidget } from '@theia/editor-preview/lib/browser';
+
+const CLOSE_ICON_WIDTH = 16;
+
+export const OPEN_EDITORS_PROPS: TreeProps = {
+    ...defaultTreeProps,
+    virtualized: false,
+    contextMenuPath: OPEN_EDITORS_CONTEXT_MENU,
+};
+@injectable()
+export class OpenEditorsWidget extends FileTreeWidget {
+    static ID = 'theia-open-editors-widget';
+    static LABEL = 'Open Editors';
+
+    static PREFIX_ICON_CLASS = 'open-editors-prefix-icon';
+
+    @inject(ApplicationShell) protected readonly applicationShell: ApplicationShell;
+
+    static createContainer(parent: interfaces.Container): Container {
+        const child = createFileTreeContainer(parent);
+
+        child.unbind(FileTreeModel);
+        child.bind(OpenEditorsModel).toSelf();
+        child.rebind(TreeModel).toService(OpenEditorsModel);
+
+        child.unbind(FileTreeWidget);
+        child.bind(OpenEditorsWidget).toSelf();
+
+        child.rebind(TreeProps).toConstantValue(OPEN_EDITORS_PROPS);
+
+        child.bind(OpenEditorsTreeDecoratorService).toSelf().inSingletonScope();
+        child.rebind(TreeDecoratorService).toService(OpenEditorsTreeDecoratorService);
+        return child;
+    }
+
+    static createWidget(parent: interfaces.Container): OpenEditorsWidget {
+        return OpenEditorsWidget.createContainer(parent).get(OpenEditorsWidget);
+    }
+
+    constructor(
+        @inject(TreeProps) readonly props: TreeProps,
+        @inject(OpenEditorsModel) readonly model: OpenEditorsModel,
+        @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer
+    ) {
+        super(props, model, contextMenuRenderer);
+    }
+
+    @postConstruct()
+    init(): void {
+        super.init();
+        this.id = OpenEditorsWidget.ID;
+        this.title.label = OpenEditorsWidget.LABEL;
+        this.addClass(OpenEditorsWidget.ID);
+        this.update();
+    }
+    protected activeTreeNodePrefixElement: string | undefined | null;
```

Comment:
The linter isn't going to like `null`, so you may need a disabling comment.

Reply:
Done 👍